docs/js/class_p.js

```diff
@@ -1688,336 +803,253 @@ class Puzzle {
+            }
+        }
+        this.make_frameline();
 
-                // Maintain Symbol elements in the same column
-                if (this[i].symbol) {
-                    let m;
-                    let temp = this[i].symbol;
-                    this[i].symbol = {};
-                    let keys = Object.keys(temp);
-                    for (var k = 0; k < keys.length; k++) {
-                        let factor = Math.floor(parseInt(keys[k]) / (originalnx0 * originalny0));
-                        m = parseInt(keys[k]) + ((parseInt((keys[k] - (factor * originalnx0 * originalny0)) / (originalnx0))) + factor * originalny0) * sign;
-                        this.record("symbol", m);
-                        this[i].symbol[m] = temp[keys[k]];
-                    }
-                }
-
-                // Maintain Line elements in the same column
-                if (this[i].line) {
-                    let m;
-                    let temp = this[i].line;
-                    this[i].line = {};
-                    for (var k in temp) {
-                        if (temp[k] === 98) {
-                            let factor = Math.floor(parseInt(k) / ((originalnx0) * (originalny0)));
-                            m = parseInt(k) + ((parseInt((parseInt(k) - (factor * originalnx0 * originalny0)) / (originalnx0))) + factor * originalny0) * sign;
-                            this.record("line", m);
-                            this[i].line[m] = temp[k];
-                        } else {
-                            let factor = Math.floor(parseInt(k.split(",")[1]) / ((originalnx0) * (originalny0)));
-                            var k1 = parseInt(k.split(",")[0]) + ((parseInt(parseInt(k.split(",")[0]) / originalnx0) - 2) + 2) * sign;
-                            if (factor == 0) {
-                                var k2 = parseInt(k.split(",")[1]) + ((parseInt(parseInt(k.split(",")[1]) / originalnx0) - 2) + 2) * sign;
-                            } else {
-                                var k2 = parseInt(k.split(",")[1]) + ((parseInt((parseInt(k.split(",")[1]) - (factor * originalnx0 * originalny0)) / (originalnx0))) + factor * originalny0) * sign;
+        this.translate_puzzle_elements(translate);
+    }
+
+    // Universal function to translate all elements in a puzzle:
+    // - cursor
+    // - selection
+    // - conflicts
+    // - all pu_q and pu_a puzzle elements
+    // - embedded soltion (single and multiple)
+    // 
+    // Currently used for resizing the board.
+    // Or in the future to insert or delete a row/col, given the proper translate function.
+    translate_puzzle_elements(translate) {
+        this.cursol = translate(this.cursol);
+        this.cursolS = translate(this.cursolS);
+        this.freelinecircle_g[0] = translate(this.freelinecircle_g[0]);
+        this.freelinecircle_g[1] = translate(this.freelinecircle_g[1]);
+        this.selection = this.selection.map(translate);

```

Comment:
It works if this.selection is a single value but fails if its in array.
It's not very critical issue but for the completion sake, it would be great if this handles arrays.

Reply:
I should clarify this.selection is an array only in Sudoku Mode.

Reply:
Ah, good catch. I will fix this tomorrow and look for other edgecases.

Reply:
I think you are mistaken, unless I am missing your point.
`this.selection` is always an array in any mode, and `this.selection.map` will correctly translate any selection.

Reply:
The variable type is an array in any mode, but it has multiple values in Sudoku Mode.
E.g., If you select the first three cells. this.selection = [a,b,c] after you apply this.selection.map it will be of size = 1.
You can try this. i.e. Select some cells. Add a row or column through resize. the selection will be reset to a single cell.

Reply:
The selection was already a single cell before entering function translate_puzzle_elements.
This happens in main.js window_click line 977 anytime you click outside the grid, except when the Ctrl key is down.

Reply:
my bad I was console logging at wrong place.
sorry about the confusion